src/bootstrap/builder.rs

```diff
@@ -968,7 +972,13 @@ impl<'a> Builder<'a> {
                 "RUSTC_DEBUGINFO_LINES",
                 self.config.rust_debuginfo_lines.to_string(),
             );
-            cargo.env("RUSTC_FORCE_UNSTABLE", "1");
+
+            if cmd == "clippy" {
```

Comment:
won't this cause everything to be rebuilt if running `./x.py build`, then `./x.py clippy` and again `./x.py build` due to the changing `RUSTFLAGS`?

Reply:
Good point, I missed that; I rebased and adjusted the PR to modify `RUSTFLAGS` in a more suitable place (I think).